### PR TITLE
Added Dockerfile and docker-compose

### LINKED ISSUE #4: Create Docker Configuration files
Add a `Dockerfile` to configure the Django application and a `docker-compose.yml` to configure the MariaDB database.

### PR DESCRIPTION
**Pull request by anirudhprabhakaran3**
# Description

Adds Dockerfiles and Docker Compose to project, so that people can develop from anywhere.

Fixes #4 and #6

## Dependencies

Docker is required on system.

## Type of Change

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

- Have run `docker compose up` on my local system, and the deploymet works properly.

---

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: corpus/corpus/settings.py [hunk 1]
```diff
@@ -20,63 +24,70 @@
...
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'django-insecure-0=$pe95skb6i4a067z-yh3#r&jtad2ar)tz9l35a5gtg(dnxy_'
+SECRET_KEY = os.environ["DJANGO_SECRET"]
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
+DEBUG = os.environ["DEBUG"] == "True"
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ["*"]
```
Comment (nishant-nayak) [inline corpus/corpus/settings.py:32]: Update this to allow only the required hosts. You can include some logic like:
```python
if DEBUG:
    ALLOWED_HOSTS = ["localhost", "127.0.0.1",]
else:
    ALLOWED_HOSTS = ["URL Of Deployment"]
```

File: corpus/corpus/settings.py [hunk 2]
```diff
@@ -20,63 +24,70 @@
...
 ]
 
-WSGI_APPLICATION = 'corpus.wsgi.application'
+WSGI_APPLICATION = "corpus.wsgi.application"
 
 
 # Database
 # https://docs.djangoproject.com/en/4.1/ref/settings/#databases
 
 DATABASES = {
```
Comment (nishant-nayak) [inline corpus/corpus/settings.py:80]: Values in this dict should not be hardcoded. Move them out into environment variables. Only the `ENGINE` can stay hardcoded.

**Review by nishant-nayak — CHANGES_REQUESTED** [review]
Add documentation in the `README.md` of this repository on the steps to set up this project using Docker Compose.

**Comment by nishant-nayak** [comment]
@anirudhprabhakaran3 the new changes are making the test build fail. Please make the necessary changes to the GitHub action as well.

**Comment by anirudhprabhakaran3** [comment]
Yup, will check 👍🏻

**Review by nishant-nayak — APPROVED** [review]
LGTM!

**Pull request merged**